### PR TITLE
Withdrawal Bug Fixes

### PR DESCRIPTION
**Pull request by ayshptk**

### CODE REVIEW

File: src/modules/app.ts [hunk 1]
```diff
@@ -344,11 +344,33 @@ export class AppModule
   public async withdrawFunds(
     to: string,
     currency: string,
-  ): Promise<TransactionReceipt> {
+  ): Promise<TransactionReceipt | void> {
```
Comment (ayshptk) [inline src/modules/app.ts:347]: @jakeloo will this be an issue?

File: src/modules/app.ts [hunk 2]
```diff
@@ -901,6 +923,29 @@ export class AppModule
...
+  }
+
+  public async upgrade(): Promise<void> {
+    if (!(await this.requireUpgrade())) {
+      return;
+    }
+    const metadata = (await this.getMetadata()).metadata;
+    const splitModule = await this.deploySplitsModule({
+      name: `${metadata?.name} Royalty Treasury`,
+      recipientSplits: [],
```
Comment (jakeloo) [inline src/modules/app.ts:945]: we want to add signer in here

File: src/modules/app.ts (hunk 1 above)
Comment (jakeloo) [reply src/modules/app.ts:347]: yes, let's make it TransactionReceipt

File: src/modules/app.ts [hunk 3]
```diff
@@ -910,7 +955,10 @@ export class AppModule
     const walletBalance = await this.readOnlyContract.provider.getBalance(
       this.address,
     );
-    return walletBalance;
+    const treasuryBalance = await this.readOnlyContract.provider.getBalance(
+      await this.getRoyaltyTreasury(),
+    );
+    return walletBalance.add(treasuryBalance);
```
Comment (jakeloo) [inline src/modules/app.ts]: need similar logic in `balanceOfToken`

File: src/modules/app.ts [hunk 4]
```diff
@@ -344,11 +344,33 @@ export class AppModule
...
+    const treasury = await this.getRoyaltyTreasury();
+    if (
+      !(await erc20.balanceOf(treasury)).isZero() &&
+      treasury !== this.address
+    ) {
+      const royalty = Royalty__factory.connect(
+        treasury,
+        this.readOnlyContract.provider,
+      );
+      return await this.sendContractTransaction(
```
Comment (jakeloo) [inline src/modules/app.ts:367]: double check the logic in splits module, i think depending on the currency token address, we need to do call differently

**Review by jakeloo — CHANGES_REQUESTED** [review]

**Pull request merged**